webapp/src/components/jira_ticket_tooltip/jira_ticket_tooltip.tsx

```diff
@@ -0,0 +1,312 @@
+import React from 'react';
+import './ticketStyle.scss';
+
+import {Dispatch} from 'redux';
+
+import {Instance} from 'types/model';
+
+export type Props = {
+    href: string;
+    show: boolean;
+    connected: boolean;
+    isLoaded?: boolean;
+    ticketDetails?: TicketDetails;
+    defaultUserInstanceID?: string;
+    connectedInstances: Instance[];
+    getIssueByKey: (issueKey: string, instanceID: string) => (dispatch: Dispatch, getState: any) => Promise<{
+        data?: TicketData;
+        error?: any;
+    }>;
+    setTicket?: (ticketDetails: {}) => void;
+}
+
+export type State = {
+    href: string;
+    isLoaded: boolean;
+    ticketId: string;
+    ticketDetails?: TicketDetails;
+};
+
+const isAssigned = ' is assigned';
+const unAssigned = 'Unassigned';
+const jiraTicketTitleMaxLength = 80;
+
+enum myStatus {
+    INDETERMINATE = 'indeterminate',
+    DONE = 'done',
+}
+
+const myStatusClasses: Record<string, string> = {
+    [myStatus.INDETERMINATE]: ' ticket-status--indeterminate',
+    [myStatus.DONE]: ' ticket-status--done',
+};
+
+export default class TicketPopover extends React.PureComponent<Props, State> {
+    truncateString(str: string, num: number) {
+        if (num > str.length) {
+            return str;
+        }
+        return `${str.substring(0, num)}...`;
+    }
```

Comment:
This can be defined as a flat function in another file

Reply:
Removed this function